gpii/node_modules/eventLog/src/eventLog.js

```diff
@@ -0,0 +1,368 @@
+        fluid.each(events, function (eventObject) {
+            gpii.eventLog.writeLog(that, eventObject.level, eventObject);
+        });
+
+        that.pendingLogs = {};
+    }
+};
+
+/**
+ * Logs an error.
+ *
+ * @param that {Component} The gpii.eventLog instance.
+ * @param moduleName {String} The part of GPII causing this error.
+ * @param errType {String} Type of error.
+ * @param err {Object} The error.
+ * @param err {Object} The error.
+ * @param level {Object} [optional] Level of the log. default: fluid.logLevel.FAIL.
+ */
+gpii.eventLog.logError = function (that, moduleName, errType, err, level) {
+    if (!level) {
+        level = fluid.logLevel.FAIL;
+    }
+
+    var data = {};
+    if (err instanceof Error) {
+        // Error doesn't serialise
+        data.error = {};
+        fluid.each(Object.getOwnPropertyNames(err), function (a) {
+            data.error[a] = err[a];
+        });
+    } else {
+        data.error = err;
+    }
+
+    var eventObject = gpii.eventLog.createLogObject(moduleName, "Error." + errType, data);
+    gpii.eventLog.writeLog(that, fluid.logLevel.FAIL, eventObject);
+};
+
+// Log uncaught exceptions.
+fluid.onUncaughtException.addListener(gpii.eventLog.gotError, "gpii-eventLog");
+
+// Log fluid.fail.
+fluid.failureEvent.addListener(function (args) {
+    var err = Array.isArray(args) ? args.join(" ") : err;
+    gpii.eventLog.gotError(err);
+}, "gpii-eventLog", "before:fail");
+
+/**
+ * Logs an error caught by onUncaughtException or failureEvent.
+ *
+ * @param err The error.
+ */
+gpii.eventLog.gotError = function (err) {
+    var eventLog = fluid.queryIoCSelector(fluid.rootComponent, "gpii.eventLog");
+    if (eventLog.length > 0) {
+        eventLog[0].logError(eventLog[0], null, "Failure", err);
+    }
+};
+
+/**
+ * Gets the path of the new log file for this instance of gpii.
+ *
+ * @param that {Component} The gpii.eventLog instance.
+ */
+gpii.eventLog.getLogFile = function (that) {
+    var logFile = that.logFilePath || process.env.GPII_EVENT_LOG;
+
+    if (!logFile) {
+        var startupTime = Date.now();
+        var gpiiSettingsDir = that.getGpiiSettingsDir();
+        logFile = path.join(gpiiSettingsDir, "gpii-" + gpii.journal.formatTimestamp(startupTime) + ".log");
+        fluid.log(fluid.logLevel.IMPORTANT, "Writing event log to " + that.logFilePath);
+    }
+
+    that.logFilePath = logFile;
+
+    return that.logFilePath;
+};
+
+/**
+ * Writes an event to the log file.
+ *
+ * @param that {Component} The gpii.eventLog instance.
+ * @param level {Object} Level of the log, see fluid.logLevelsSpec [FATAL,FAIL,WARN,IMPORTANT,INFO,TRACE].
+ * @param event {Object} The object. This will be modified to what has been sent to the log.
+ */
+gpii.eventLog.writeLog = function (that, level, event) {
+    var intLevel = gpii.eventLog.checkLevel(level);
+    event.level = intLevel.value;
+
+    fluid.logObjectRenderChars = 0xffffff;
```

Comment:
Setting this configuration without restoring it is what triggers the Infusion log rendering bug which causes the overall test failure - we attempt to render for logging a node "IncomingMessage" which has a getter for "encrypted" which can throw.